python/meterpreter/tests/test_ext_server_stdapi.py

```diff
@@ -0,0 +1,277 @@
+        mock_popen.return_value = process_mock
+
+        result = self.ext_server_stdapi[
+            "stdapi_net_config_get_interfaces_via_osx_ifconfig"
+        ]()
+
+        expected = [
+            {
+                "addrs": [
+                    (
+                        socket.AF_INET,
+                        b"\xc0\xa8\x01\xa6",
+                        b"\xff\xff\xff\x00",
+                    )
+                ],
+                "flags": 8863,
+                "flags_str": "UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST",
+                "hw_addr": '\x11"3DUf',
+                "index": 0,
+                "mtu": 1500,
+                "name": "en0",
+            }
+        ]
+
+        self.assertEqual(result, expected)
+
+    @mock.patch("subprocess.Popen")
+    def test_stdapi_sys_process_get_processes_via_ps(self, mock_popen):
+        command_result = b"""
+  PID  PPID USER             COMMAND
+    1     0 root             /sbin/launchd
+   88     1 root             /usr/sbin/syslogd
+   89     1 root             /usr/libexec/UserEventAgent (System)
+""".lstrip()
+
+        process_mock = mock.Mock()
+        attrs = {
+            "communicate.return_value": (command_result, b""),
+            "wait.return_value": ERROR_SUCCESS,
+        }
+        process_mock.configure_mock(**attrs)
+        mock_popen.return_value = process_mock
+
+        request = bytes()
+        response = bytes()
+        result = self.ext_server_stdapi["stdapi_sys_process_get_processes_via_ps"](
+            request, response
+        )
+
+        self.assertErrorSuccess(result)
+
+
+class ExtServerStdApiFileSystemTest(ExtServerStdApiTest):
+    def test_stdapi_fs_stat(self):
+        request = bytes()
+        request += self.meterpreter_context["tlv_pack"](
+            self.ext_server_stdapi["TLV_TYPE_FILE_PATH"],
+            os.path.dirname(os.path.abspath(__file__)),
+        )
+        response = bytes()
+        self.assertMethodErrorSuccess("stdapi_fs_stat", request, response)
+
+    # Older versions of Python on Windows return invalid/negative values for st_rdev
+    # https://github.com/python/cpython/commit/a10c1f221a5248cedf476736eea365e1dfc84910#diff-b419a047f587ec3afef8493e19dbfc142624bf278f3298bfc74729abd89e311d
+    @mock.patch("os.stat")
+    @mock.patch("sys.platform")
+    def test_stdapi_fs_stat_with_negative_st_rdev_on_windows(
+        self, mock_sys_platform, mock_os_stat
+    ):
+        os_stat_result = mock.MagicMock()
+        os_stat_result.configure_mock(
+            **{
+                "st_mode": 33206,
+                "st_ino": 281474976726344,
+                "st_dev": 3323847249,
+                "st_nlink": 1,
+                "st_uid": 0,
+                "st_gid": 0,
+                "st_size": 9884,
+                "st_rdev": -1910224650,
+                "st_atime": 1686079301.2200336,
+                "st_mtime": 1686079301.2200336,
+                "st_ctime": 1686079301.2200336,
```

Comment:
Somewhat related - `st_ctime` is deprecated now in Python 3.12, and replaced with birthtime:

>     The ``st_ctime`` attribute of a stat result is deprecated on Windows.
>      The file creation time is properly available as ``st_birthtime``, and
>      in the future ``st_ctime`` may be changed to return zero or the
>      metadata change time, if available.


I can't imagine them removing `st_ctime`, but for now it's copied acros:

```
    /* ctime is only deprecated from 3.12, so we copy birthtime across */
    result->st_ctime = result->st_birthtime;
    result->st_ctime_nsec = result->st_birthtime_nsec;
```

The automated CI tests will catch the scenario of them ever removing this method, or I can future proof this now if we think it's best to do it whilst the knowledge is fresh